File: Arrays/number-of-zero-filled-subarrays/number-of-zero-filled-subarrays.go
// https://leetcode.com/problems/number-of-zero-filled-subarrays/

package main

func zeroFilledSubarray(nums []int) int64 {
  // Arithmetric Progression Sum for each sequence of zeroes
  res := int64(0)
  count := int64(0)

  for _, num := range nums {
    if num == 0 {
      count++
    } else {
      res += (count * (count + 1)) / 2
      count = 0
    }
  }
  res += (count * (count + 1)) / 2
  return res
}

File: Arrays/number-of-zero-filled-subarrays/number-of-zero-filled-subarrays.ts
// https://leetcode.com/problems/number-of-zero-filled-subarrays/

function zeroFilledSubarray(nums: number[]): number {
  // Arithmetric Progression Sum for each sequence of zeroes
  let res = 0;
  let zeroCount = 0;
  const length = nums.length;

  for (let i = 0; i < length; i++) {
    if (nums[i] == 0) {
      zeroCount++;
    } else {
      res += ((1 + zeroCount) * zeroCount) / 2;
      zeroCount = 0;
    }
  }
  if (zeroCount > 0) {
    res += ((1 + zeroCount) * zeroCount) / 2;
  }

  return res;
}
